src/formatDate.js

```diff
@@ -50,7 +50,48 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const symbolA = fromFormat[fromFormat.length - 1];
+  const symbolB = toFormat[toFormat.length - 1];
+
+  const newDateArray = date.split(symbolA).join(symbolB).split(symbolB);
+  const newToFormat = toFormat.filter(dateExpression =>
+    dateExpression !== dateExpression[dateExpression.length - 1]);
+  const newFromFormat = fromFormat.filter(dateExpression =>
+    dateExpression !== dateExpression[dateExpression.length - 1]);
+
+  const obj = {};
+  const objFormatted = {};
+
+  for (let i = 0; i < newFromFormat.length; i++) {
+    obj[newFromFormat[i]] = newDateArray[i];
+  }
+
+  for (let i = 0; i < newToFormat.length; i++) {
+    for (let j = 0; j < newFromFormat.length; j++) {
+      const newDateExpression = newToFormat[i];
+      const oldDateExpression = newFromFormat[j];
+
+      if (newDateExpression === 'DD') {
+        objFormatted['DD'] = obj['DD'];
+      } else if (newDateExpression === 'MM') {
+        objFormatted['MM'] = obj['MM'];
+      } else if (newDateExpression === 'YY') {
+        objFormatted['YY'] = obj['YYYY'].slice(2);
+      } else if (newDateExpression === 'YYYY' && oldDateExpression === 'YYYY') {
+        objFormatted['YYYY'] = obj['YYYY'];
+      } else if (newDateExpression === 'YYYY' && oldDateExpression === 'YY') {
+        if (obj['YY'] < 30) {
+          objFormatted['YYYY'] = obj['YY'].padStart(4, '20');
+        } else {
+          objFormatted['YYYY'] = obj['YY'].padStart(4, '19');
+        }
+      }
+    }
+  }
+
+  const result = Object.values(objFormatted).join(symbolB);
```

Comment:
Remove the unnecessary variable 'result' and directly return the formatted date string: return Object.values(formattedDateObj).join(symbolB)